examples/common/src/framework.rs

```diff
@@ -290,116 +292,108 @@ fn start<E: Example>(
     let (mut frame_count, mut accum_time) = (0, 0.0);
 
     log::info!("Entering render loop...");
-    event_loop.run(move |event, _, control_flow| {
+    event_loop.run(move |event, target| {
         let _ = (&instance, &adapter); // force ownership by the closure
-        *control_flow = if cfg!(feature = "metal-auto-capture") {
-            ControlFlow::Exit
-        } else {
-            ControlFlow::Poll
+        target.set_control_flow(ControlFlow::Poll);
+
+        if cfg!(feature = "metal-auto-capture") {
+            target.exit();
         };
-        match event {
-            event::Event::RedrawEventsCleared => {
-                #[cfg(not(target_arch = "wasm32"))]
-                spawner.run_until_stalled();
 
-                window.request_redraw();
-            }
+        match event {
             event::Event::WindowEvent {
-                event:
-                    WindowEvent::Resized(size)
-                    | WindowEvent::ScaleFactorChanged {
-                        new_inner_size: &mut size,
-                        ..
-                    },
+                event: WindowEvent::Resized(size),
                 ..
-            } => {
-                log::info!("Resizing to {:?}", size);
-                config.width = size.width.max(1);
-                config.height = size.height.max(1);
-                example.resize(&config, &device, &queue);
-                surface.configure(&device, &config);
+            } => resize_window(&mut example, size, &mut config, &mut surface, &device, &queue),
+            event::Event::WindowEvent {
+                event: WindowEvent::ScaleFactorChanged { scale_factor, .. },
```

Comment:
This event could be ignored, as the resized event will be triggered anyway and this would resize two times

Reply:
good point! Thank you for pointing this out!

Reply:
removed!